src/Uno.UI/UI/Xaml/Controls/Repeater/ViewManager.cs

```diff
@@ -0,0 +1,984 @@
+		{
+			UIElement element = null;
+
+			// See if you can find something among the pinned elements.
+			for (int i = 0; i < m_pinnedPool.Count; ++i)
+			{
+				var elementInfo = m_pinnedPool[i];
+				var virtInfo = elementInfo.VirtualizationInfo;
+
+				if (virtInfo.Index == index)
+				{
+					m_pinnedPool.RemoveAt(i);
+					element = elementInfo.PinnedElement;
+					elementInfo.VirtualizationInfo.MoveOwnershipToLayoutFromPinnedPool();
+
+					// Update realized indices
+					m_firstRealizedElementIndexHeldByLayout = Math.Min(m_firstRealizedElementIndexHeldByLayout, index);
+					m_lastRealizedElementIndexHeldByLayout = Math.Max(m_lastRealizedElementIndexHeldByLayout, index);
+					break;
+				}
+			}
+
+			return element;
+		}
+
+		// There are several cases handled here with respect to which element gets returned and when DataContext is modified.
+		//
+		// 1. If there is no ItemTemplate:
+		//    1.1 If data is a UIElement . the data is returned
+		//    1.2 If data is not a UIElement . a default DataTemplate is used to fetch element and DataContext is set to data**
+		//
+		// 2. If there is an ItemTemplate:
+		//    2.1 If data is not a FrameworkElement . Element is fetched from ElementFactory and DataContext is set to the data**
+		//    2.2 If data is a FrameworkElement:
+		//        2.2.1 If Element returned by the ElementFactory is the same as the data . Element (a.k.a. data) is returned as is
+		//        2.2.2 If Element returned by the ElementFactory is not the same as the data
+		//                 . Element that is fetched from the ElementFactory is returned and
+		//                    DataContext is set to the data's DataContext (if it exists), otherwise it is set to the data itself**
+		//
+		// **data context is set only if no x:Bind was used. ie. No data template component on the root.
+		UIElement GetElementFromElementFactory(int index)
+		{
+			// The view generator is the provider of last resort.
+			var  data = m_owner.ItemsSourceView.GetAt(index);
+
+			UIElement GetElement()
+			{
+				var elementFactory = m_owner.ItemTemplateShim;
+				if (elementFactory == null)
+				{
+					if (data is UIElement dataAsElement)
+					{
+						return dataAsElement;
+					}
+					else
+					{
+						// If no ItemTemplate was provided, use a default
+						//var factory  = XamlReader.Load("<DataTemplate xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'><TextBlock Text='{Binding}'/></DataTemplate>") as DataTemplate;
+						var factory = new DataTemplate(() =>
+						{
+							var tb = new TextBlock();
+							tb.SetBinding(TextBlock.TextProperty, new Binding());
+							return tb;
+						});
+						m_owner.ItemTemplate = factory;
+
+						elementFactory = m_owner.ItemTemplateShim;
+					}
+				}
+
+				if (m_ElementFactoryGetArgs == null)
+				{
+					m_ElementFactoryGetArgs = new ElementFactoryGetArgs();
+				}
+
+				var args = m_ElementFactoryGetArgs;
+
+				using var scopeGuard = Disposable.Create(() =>
```

Comment:
Codacy found an issue: [Remove the unused local variable 'scopeGuard'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5863680)